### DISCUSSION TITLE
workflow queries

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#786] [Questions / Documentation]
Hi,
I have a few workflow questions:

1. Do you recommend iterative clustering based on variable features before or after batch corrections? In general, I have seen that clustering based on top features followed by batch corrections seems more appropriate than iterative clustering with top followed by variable features.
2. Also, would it reasonable to do clustering based on top features only and then using those clusters for iterative sub-clustering with variable features? 
3. Harmony corrections do not always integrate batches properly (with default parameters). Do you recommend any specific parameter tuning in harmony for more stringent batch-correction?
4. How can I use harmony-corrected LSI dimensions for iterative sub-clustering of clusters?
5. Do I used harmony-correct LSI or IterativeLSI for cell-type annotation?

Thanks a lot for your help!

**Comment by rcorces** [comment]
Please note that Questions and Feature Requests belong in Discussions. I have migrated your issue there.

1) I'm not sure what you mean by "iterative clustering" but if you mean iterative LSI then yes. But if you are using harmony for batch effect correction then you should call clusters on the harmony space.
2) Again, I think you are confusing dimensionality reduction and clustering and it is difficult to interpret your question.
3) No recommendations beyond what is suggested in the manual or in the documentation for the harmony package.
4&5) Please read the manual and attempt to refine your questions as I'm having difficulty understanding.

**Comment by AnjaliC4** [comment]
Hi Ryan, thanks! My apologies for not being very clear. I will try reframing my question.
In short, we are seeing strong batch effects (due to different techs & batches) that are not corrected by iterative LSI. I tried harmony with the default setting to correct them using "batch" as a variable, however, I still see some batch effects. So these were my steps
1. LSI (with 1 iteration) only on top most accessible features (iterative LSI and clustering using top+variable features before harmony performed worse for batch corrections- not sure why. Would appreciate your input.
2. Harmony correction
3. Clustering (0.5 resolution) in harmony space
4. UMAP
With these steps, all the clusters in the UMAP plot were very close to each other (i.e. not enough separation between them) and some clusters (c15,c24) were entirely driven by one batch (because harmony didn't seem to correct batch differences entirely).
I have added an example of what I mean. Please do let me know if something is still unclear and please let me know if you have suggestions. Thanks so much!
[Plot-UMAP2Harmony.pdf](https://github.com/GreenleafLab/ArchR/files/6565461/Plot-UMAP2Harmony.pdf)

**Reply by rcorces** [reply]
I think the best solution would be to change how you are using Harmony. Perhaps you are already aware and have tried this but you can correct for multiple covariates (for example, batch and tech in the example you mentioned). See the section "Harmony with two or more covariates" on this page - https://github.com/immunogenomics/harmony

Its also worth noting that not all batch effects can be removed and if you have very strong effects then there is only so much that the software can do.
